src/hexdoc/_templates/components/table_of_contents.html.jinja

```diff
@@ -26,3 +26,18 @@
     </details>
   {% endfor %}
 </nav>
+
+<script>
+// Add this script right before the closing body tag or in a separate JS file
+document.addEventListener('DOMContentLoaded', (event) => {
+  const btn = document.getElementById('collapse-button');
+  const tocContainer = document.querySelector('.toc-container');
+
+  btn.addEventListener('click', function() {
+    // Toggle the ToC container visibility
+    tocContainer.style.display = tocContainer.style.display === 'none' ? 'block' : 'none';
```

Comment:
I think we have a `.hidden` class already, right? not sure if that works here though